src/Glpi/Inventory/Asset/Device.php

```diff
@@ -197,4 +198,14 @@ protected function itemdeviceAdded(Item_Devices $itemdevice, $val)
     {
        //to be overrided
     }
+
+    public function checkConf(Conf $conf): bool
+    {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+        /** @var \Item_Devices $item_device */
+        $item_device = $this->getItemtype();
+        $cfg_key = 'item' . strtolower($item_device::getDeviceType()) . '_types';
+        return in_array($this->item->getType(), $CFG_GLPI[$cfg_key]);
```

Comment:
The `itemAffinity()` method of the `Item_Devices` concrete implementation is more suitable as it will fallback to the default `itemdevices_itemaffinity` configuration if the device has no proper configuration entry.

```suggestion
        return in_array($this->item->getType(), $item_device::itemAffinity());
```